src/includes/traits/Plugin/AutoCacheUtils.php

```diff
@@ -170,5 +170,49 @@ public function autoCacheCheckXmlSitemap($sitemap, $is_nested_sitemap = false, $
 
         return true;
     }
+
+    /**
+     * Download XML Sitemap via cURL and store in a temp file.
+     *
+     * @since 16xxxx
+     *
+     * @param string $url to XML Sitemap that needs to be downloaded
+     *
+     * @throws \Exception If unable to write temporary file or download XML Sitemap
+     *
+     * @return string Path to downloaded XML Sitemap file
+     */
+    public function autoCacheGetSitemapViaCurl($url)
+    {
+        if (!($tmp_dir = $this->getTmpDir())) {
+            throw new \Exception(__('No writable tmp directory.', SLUG_TD));
+        }
+
+        $tmp_file = $tmp_dir.'/'.$this->addTmpSuffix($tmp_dir).'.xml';
```

Comment:
@raamdev Lookin' great!

On this line _(this is an inline code comment)_... the `addTmpSuffix()` call is probably not what you want because that function is designed to add a tmp suffix to an existing directory for the purpose of working on a temporary copy of an existing directory; e.g., renaming the cache directory and working on a temporary copy while the cache is locked. So `addTmpSuffix()` will return the full absolute path, but with a `-tmp` suffix.

In the use case you have here, the full path to the `$tmp_file` is probably not what you expect it to be.

You might be better off using: https://developer.wordpress.org/reference/functions/wp_tempnam/